### ISSUE TITLE
AWS EKS Version Upgrade using Terraform

### ISSUE THREAD

**Issue by dhineshbabuelango** [terraform-providers/terraform-provider-aws#9878]
Hi Team, Currently is there any way we can use terraform to upgrade the EKS cluster version, which is basically equal to the below AWS command.

aws eks update-cluster-version --name Your-EKS-Cluster --kubernetes-version 1.11

**Comment by ewbankkit** [comment]
@dhineshbabuelango The `aws_eks_cluster`'s [`version` attribute](https://www.terraform.io/docs/providers/aws/r/eks_cluster.html#version) provides the functionality you are looking for. A change of value triggers a call to the [EKS `UpdateClusterVersion` API](https://docs.aws.amazon.com/eks/latest/APIReference/API_UpdateClusterVersion.html).

**Comment by dhineshbabuelango** [comment]
Thanks Kit, We tried updating the version, everything is working as expected. But because of we move the EKS version, we also tried to update the respective AMI for worker nodes to autoscaling group even that went fine. but terraform just changed the autoscaling group configuration and doesn't terminated the existing instance instead the worker nodes were still using the old AMI. When we manually terminated the worker node instances, autoscaling group provisioned the worker nodes using the new AMI. is this the expected behaviour?
